### PR TITLE
A package to query openai (chatGPT)

### PR DESCRIPTION
**Pull request by rohitna**
### Clipboard queries
- `:ask-gpt` (queries the content of the clipboard)
- `:rephrase` (rephrases the content of the clipboard)
- `:correct` (corrects grammatical errors in the contents of the clipboard)
- `:fact-check` (fact checks the content of the clipboard)
- `:respond` (writes a response to the content of the clipboard)
- `:explain` (explain the content of the clipboard in detail)
- `:eli5` (explain the clipboard content to me like I am 5)
- `:summarize` (summarize the content of the clipboard)
- `:write` (write an article on the content of the clipboard)
- `:debug` (debug the code in the clipboard)
- `:grade` (grade the content of the clipboard)

### Regex query
- `:Q/{query}//` (regex query, char limit 25)
  - Example: `:Q/rust quicksort//` returns the following:
  ```rust
    fn quicksort(vec: &mut [i32]) {
        if vec.len() < 2 {
            return;
        }
        let pivot = vec.len() - 1;
        let mut pos = 0;
        for i in 0..pivot {
            if vec[i] <= vec[pivot] {
                vec.swap(i, pos);
                pos += 1;
            }
        }
        vec.swap(pivot, pos);
        quicksort(&mut vec[..pos]);
        quicksort(&mut vec[pos+1..]);
    }
    ```

### CODE REVIEW

**Comment by federico-terzi** [comment]
@rohitna Looks amazing! Could we maybe add a warning in the readme saying that this requires Bash+Curl, and thus might not work on Windows?

**Comment by federico-terzi** [comment]
We also need to add the "author" field to the manifest, as reported by the CI :)

**Comment by rohitna** [comment]
Actually, while working with it I realized that it errors out whenever the prompts have quotes or apostrophes. I found it quite tricky to handle non-strict json data, like code snippets, just using bash. I wrote a more flexible version but with python dependencies here, please check it out: https://github.com/rohitna/chatgpt-script

I am still working on adding more features to it, and also asking people to test it before thinking about creating a new Espanso package out of it.

Perhaps, it is best to close this PR as this package doesn't work very well and is also not context aware.

**Comment by federico-terzi** [comment]
Sounds good, feel free to reopen when ready :)
